### PR TITLE
[tests-only][full-ci] Forwardport added test coverage for previewing shared resources using spaces dav path

### PR DESCRIPTION
**Pull request by PrajwolAmatya**
## Description
This PR forward port: https://github.com/owncloud/ocis/pull/7234

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [x] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

File: tests/acceptance/features/coreApiWebdavPreviews/previews.feature [hunk 1]
```diff
@@ -246,19 +249,20 @@ Feature: previews of files downloaded through the webdav API
...
-    And as user "Brian" the preview of "Shares/FOLDER/lorem.txt" with width "32" and height "32" should have been changed
+    And as user "Brian" the preview of shared resource "Shares/FOLDER/lorem.txt" with width "32" and height "32" should have been changed
     Examples:
-      | dav-path-version |
-      | old              |
-      | new              |
+      | dav-path-version | http-status-code |
+      | old              | 200              |
+      | new              | 200              |
+      | spaces           | 204              |
```
Comment (saw-jan) [inline tests/acceptance/features/coreApiWebdavPreviews/previews.feature:265]: I wouldn't expect to different codes for different paths
```suggestion
      | old              | 204              |
      | new              | 204              |
      | spaces           | 204              |
```

**Review by saw-jan — CHANGES_REQUESTED** [review]

File: tests/acceptance/features/bootstrap/WebDav.php [hunk 2]
```diff
@@ -4949,15 +5139,9 @@ public function theDownloadedPreviewContentShouldMatchWithFixturesPreviewContent
...
-			$path,
-			null,
-			$width,
-			$height
-		);
-		$this->theHTTPStatusCodeShouldBe(200, "", $response);
-		$this->imageDimensionsShouldBe($width, $height, $response);
+		$response = $this->downloadPreviewOfFiles($user, $path, $width, $height);
+		$this->theHTTPStatusCodeShouldBe(200, '', $response);
+		$this->imageDimensionsShouldBe($width, $height);
```
Comment (saw-jan) [inline tests/acceptance/features/bootstrap/WebDav.php:5144]: Existing one is correct I guess
```suggestion
```

File: tests/acceptance/features/bootstrap/WebDav.php [hunk 3]
```diff
@@ -4973,15 +5157,9 @@ public function userDownloadsThePreviewOfWithWidthAndHeight(string $user, string
...
-			$path,
-			null,
-			$width,
-			$height
-		);
-		$this->theHTTPStatusCodeShouldBe(200, "", $response);
-		$newResponseBodyContents = $response->getBody()->getContents();
+		$response = $this->downloadPreviewOfFiles($user, $path, $width, $height);
+		$this->theHTTPStatusCodeShouldBe(200, '', $response);
+		$newResponseBodyContents = $this->response->getBody()->getContents();
```
Comment (saw-jan) [inline tests/acceptance/features/bootstrap/WebDav.php:5162]: Existing one is correct I guess
```suggestion
```

File: tests/acceptance/features/coreApiWebdavPreviews/previews.feature (hunk 1 above)
Comment (PrajwolAmatya) [reply tests/acceptance/features/coreApiWebdavPreviews/previews.feature:265]: The test failed for old and new dav versions giving the status code `200`. So updated the status code.

Comment (saw-jan) [reply tests/acceptance/features/coreApiWebdavPreviews/previews.feature:265]: But we should not. Upload to existing file should always give `204`. I found some possible issues in webdav.php. Maybe that was causing this.

File: tests/acceptance/features/bootstrap/WebDav.php (hunk 2 above)
Comment (PrajwolAmatya) [reply tests/acceptance/features/bootstrap/WebDav.php:5144]: This could be the reason for different response status code.

**Review by saw-jan — APPROVED** [review]
LGTM :+1:

**Pull request merged**